cli/main.rs

```diff
@@ -178,114 +181,273 @@ fn print_cache_info(state: &GlobalState, json: bool) -> Result<(), ErrBox> {
+    total_sizes: &mut HashMap<String, usize>,
+    graph: &ModuleGraph,
+    root: &ModuleSpecifier,
+  ) -> Self {
+    let name = root.to_string();
+    let never_seen = seen.insert(name.clone());
+    let file = graph.get(&name).unwrap();
+    let size = file.size();
+    let deps = if never_seen {
+      file
+        .imports
+        .iter()
+        .map(|import| import.resolved_specifier.clone())
+        .map(|spec| Self::cstr_helper(seen, total_sizes, graph, &spec))
+        .collect::<Vec<_>>()
+    } else {
+      vec![]
+    };
+
+    let total_size = if never_seen {
+      if let Some(total_size) = total_sizes.get(&name) {
+        Some(total_size.to_owned())
+      } else {
+        let total: usize = deps
+          .iter()
+          .map(|dep| {
+            if let Some(total_size) = dep.total_size {
+              total_size
+            } else {
+              0
+            }
+          })
+          .sum();
+        let total = size + total;
+
+        total_sizes.insert(name.clone(), total);
+
+        Some(total)
+      }
+    } else {
+      None
+    };
+
+    Self {
+      name,
+      size,
+      total_size,
+      deps,
+    }
+  }
+}
+
+/// Prints the provided `ModuleSpecifier` file info.
 async fn print_file_info(
   worker: &MainWorker,
   module_specifier: ModuleSpecifier,
   json: bool,
 ) -> Result<(), ErrBox> {
   let global_state = worker.state.borrow().global_state.clone();
-
-  let out = global_state
-    .file_fetcher
+  let ts_compiler = &global_state.ts_compiler;
+  let file_fetcher = &global_state.file_fetcher;
+  let out = file_fetcher
     .fetch_source_file(&module_specifier, None, Permissions::allow_all())
     .await?;
-
-  let mut output = FileInfoOutput {
-    local: out.filename.to_str().unwrap(),
-    file_type: msg::enum_name_media_type(out.media_type),
-    compiled: None,
-    map: None,
-    deps: None,
-  };
-
-  let module_specifier_ = module_specifier.clone();
-
-  global_state
-    .prepare_module_load(
-      module_specifier_.clone(),
-      None,
-      TargetLib::Main,
-      Permissions::allow_all(),
-      false,
-      global_state.maybe_import_map.clone(),
-    )
-    .await?;
-  global_state
-    .clone()
-    .fetch_compiled_module(module_specifier_, None)
-    .await?;
-
-  if out.media_type == msg::MediaType::TypeScript
-    || (out.media_type == msg::MediaType::JavaScript
-      && global_state.ts_compiler.compile_js)
-  {
-    let compiled_source_file = global_state
-      .ts_compiler
-      .get_compiled_source_file(&out.url)
-      .unwrap();
-    output.compiled =
-      compiled_source_file.filename.to_str().map(|s| s.to_owned());
-  }
-
-  if let Ok(source_map) = global_state
-    .clone()
-    .ts_compiler
+  let local = out.filename.to_str().unwrap();
+  let file_type = msg::enum_name_media_type(out.media_type);
+  let compiled: Option<String> = ts_compiler
+    .get_compiled_source_file(&out.url)
+    .ok()
+    .and_then(get_source_filename);
```

Comment:
Calling this method before `global_state.prepare_module_load` will not run module graph through TS compiler, meaning if it's first time checking the file `compiled` field will not show up

Reply:
Ahh ok. I will add a test for this case and fix it.